### PR TITLE
bump kube-state-metrics to v1.0.3

### PR DESCRIPTION
**Pull request by wtty-fool**
Towards https://github.com/giantswarm/giantswarm/issues/8569

### CODE REVIEW

File: azure.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@
       version: 1.2.0
     - app: kube-state-metrics
       componentVersion: 1.9.2
-      version: 1.0.2
+      version: 1.0.3
```
Comment (sslavic) [inline azure.yaml]: IMO this shouldn't be done - changing the definition of already published release.

@wtty-fool would it make sense to keep this PR only for adding 1.0.3 kube-state-metrics to the apps changelog? We're in the process of releasing next patch releases (see other PRs in the repo), would be happy to include kube-state-metrics v1.0.3 in them

File: kvm.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@
       version: 1.1.3
     - app: kube-state-metrics
       componentVersion: 1.9.2
-      version: 1.0.2
+      version: 1.0.3
```
Comment (sslavic) [inline kvm.yaml]: This is also published release, they shouldn't be modified.

File: azure.yaml (hunk 1 above)
Comment (kopiczko) [reply azure.yaml]: Thanks @sslavic for spotting that. Then we can include that only for AWS as this is apparently the only provider with WIP release in place... Is that ok?

Comment (sslavic) [reply azure.yaml]: Sure, once you merge I'll take it into separate already created PR which activates that release and makes more changes to the release definition/package to ship them all together as one.

Comment (wtty-fool) [reply azure.yaml]: @sslavic Good catch, thank you for pointing that out.
Ok, I'll keep an update to AWS and changelog, the rest will be dropped from this PR.

File: app/changelog/kube-state-metrics.yaml [hunk 3]
```diff
@@ -1,3 +1,9 @@
+- version: 1.0.3
+  componentVersion: 1.9.2
+  description: Remove CPU and memory limits.
+  kind: fixed
```
Comment (sslavic) [inline app/changelog/kube-state-metrics.yaml]:
```suggestion
  kind: changed
```

Comment (sslavic) [reply app/changelog/kube-state-metrics.yaml]: If for nothing else, then for consistency with equivalent changelog entries done for other apps before.

**Pull request merged**